build/build.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+git_root="$(git rev-parse --show-toplevel)"
+
+: "${VERSION:="$("${git_root}/third-party/kubecf-tools/versioning/versioning.rb")"}"
+: "${IMAGE_NAME:=splatform/mits}"
+: "${IMAGE_TAG:="${IMAGE_NAME}:${VERSION}"}"
+: "${OUTPUT_CHARTS_DIR:="${git_root}/output"}"
+: "${TMP_BUILD_DIR:="${git_root}/tmp"}"
+: "${CHART_SRC:="${git_root}/chart/mits"}"
+
+################################################################################
+# BUILD IMAGE
```

Comment:
Since the two major parts are mostly unrelated, maybe we can have one target to build the chart, and a different one to build the image? (Mostly so that we might use [GitHub Actions](https://github.com/docker/build-push-action) to do the image build part, which lets us embed things like `org.opencontainers.image.source` labels.)

Reply:
I don't like the `docker/build-push-action` because it deviates from how we determine the image full tag and how we interpolate it in the chart when building. This also makes me push back this comment entirely because having them be built in the same script helps with not getting out of sync when building the chart.

Reply:
A new world opened in front of me: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable.
I was not aware of that!
I'll check the whole code for how I can improve it using these features.